### PR TITLE
feat(model): add Allocator class for performant allocations

### PR DESCRIPTION
**Pull request by anarchuser**
It has its own internal heap which it uses to allocate objects next to each other, as opposed to `new` which possibly allocates objects randomly in memory.

depends on #39

### CODE REVIEW

**Comment by anarchuser** [comment]
Does not have a concrete implementation yet

**Comment by anarchuser** [comment]
Files `src/model/allocator/allocator.h` and `test/model/allocator/allocator.cpp` can already be reviewed; they're independent from #39 and from any future implementation of it

File: src/config.h [hunk 1]
```diff
@@ -1,6 +1,8 @@
 #ifndef EVE_CONFIG_H
 #define EVE_CONFIG_H
 
+#define AMOUNT_ALLOCATED_OBJECTS 10000
```
Comment (xeruf) [inline src/config.h]:
```suggestion
#define PREALLOCATED_OBJECTS 10000
```

File: src/model/object/entity/individual/individual.h [hunk 2]
```diff
@@ -26,7 +26,7 @@ class Individual : public Entity {
     [[nodiscard]] Vector getVision() const;
 
     void turnBy (Angle angle);
-
+
```
Comment (xeruf) [inline src/model/object/entity/individual/individual.h]: try to avoid random reformatting ;)

File: test/model/allocator/allocator.cpp [hunk 3]
```diff
@@ -0,0 +1,60 @@
+#include "../../../src/model/allocator/allocator.h"
+
+#include <catch2/catch.hpp>
+
+TEST_CASE ("I can allocate objects always located close to each other") {
```
Comment (xeruf) [inline test/model/allocator/allocator.cpp]:
```suggestion
TEST_CASE ("Jointly allocated objects are located close to each other") {
```

**Review by xeruf — CHANGES_REQUESTED** [review]

**Review by xeruf — APPROVED** [review]

**Pull request merged**